InvenTree/part/test_api.py

```diff
@@ -403,6 +404,54 @@ class Target(IntEnum):
                     child.refresh_from_db()
                     self.assertEqual(child.parent, parent_category)
 
+    def test_structural(self):
+        """Test the effectiveness of structural categories
+
+        Make sure:
+        - Parts cannot be created in structural categories
+        - Parts cannot be assigned to structural categories
+        """
+        structural_category = PartCategory.objects.create(
+            name='Structural category',
+            description='This is the structural category',
+            parent=None,
+            structural=True
+        )
+
+        part_count_before = Part.objects.count()
+        try:
+            part = Part.objects.create(
+                name="Part which shall not be created",
+                description="-",
+                category=structural_category
+            )
+        except ValidationError:
```

Comment:
Suggestion: A better way to make sure the right error is thrown is with wrapping the whole thing in an assertion instead of an try/except

```
with self.assertRaises(ValidationError):
    …
```